File: 975-range-sum-of-bst/range-sum-of-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int rangeSumBST(TreeNode* root, int low, int high) {
        int add_num = 0;
        if (root == nullptr){
            return 0;
        }

        int left_sum = rangeSumBST(root->left, low, high);
        int right_sum = rangeSumBST(root->right, low, high);

        if (root->val >= low && root->val <= high) {
            add_num = root->val;
        }
        

        return add_num + left_sum + right_sum;
        
    }
};

File: 975-range-sum-of-bst/range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        if root is None:
            return 0


        left_sum = self.rangeSumBST(root.left, low, high) 
        right_sum = self.rangeSumBST(root.right, low, high) 

        add_num = root.val if root.val >= low and root.val <= high else 0

        return add_num + left_sum + right_sum
        